include/envoy/http/codec.h

```diff
@@ -148,11 +148,24 @@ class ConnectionCallbacks {
 };
 
 /**
- * A list of options that can be specified when creating a codec.
+ * http/2 codec settings
  */
-class CodecOptions {
-public:
-  static const uint64_t NoCompression = 0x1;
+struct Http2Settings {
```

Comment:
My personal preference would be to make this a class that hides a bit of it's own logic - I touched on this a bit in another comment.  It would enable writing code like calling "http2settings.use_compression()" to find out if compression is enabled rather than requiring the caller to understand that codec_options_ is really a bit field.